packages/typespec-azure-resource-manager/test/rules/arm-delete-response-codes.test.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  BasicTestRunner,
+  LinterRuleTester,
+  createLinterRuleTester,
+} from "@typespec/compiler/testing";
+import { beforeEach, describe, it } from "vitest";
+import { armDeleteResponseCodesRule } from "../../src/rules/arm-delete-response-codes.js";
+import { createAzureResourceManagerTestRunner } from "../test-host.js";
+
+describe("typespec-azure-resource-manager: arm delete response codes rule", () => {
```

Comment:
we don't need a top level describe anymore with vitest as everything is already grouped per file

Reply:
I'd prefer to keep it consistent with other files.

Reply:
Changing all the existing test will cause massive diff so it’s not really worth it I think unless refactoring significantly a test file. But here for new files this just adds extra layer that make the cli and test explorer a bit worse

Reply:
I filed this issue to restore support for mocha https://github.com/Azure/typespec-azure/issues/302.

The describes grouping remains useful when using that framework.

Reply:
mocha is vastly inferior in the capacity it gives us in test, we also were using completely outdated packages that had vulnerability, reverting is most likely not going to happen. apart from the extension issue vitest is better in every way. vitest is a modern alternative to `jest` which was the most popular test framework but we switch right at the time they got to v1 and the extension wasn't really ready but they have been pushing many updates in the last weeks so this feels uncessary to revert just because they have a bug they will most likely fix.